dagger/main.go

```diff
@@ -0,0 +1,503 @@
+package main
+
+import (
+	"context"
+	encodingjson "encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"dagger.io/dagger"
+	"github.com/knadh/koanf/providers/env"
+	"github.com/knadh/koanf/providers/posflag"
+	"github.com/knadh/koanf/providers/structs"
+	"github.com/knadh/koanf/v2"
+	"github.com/mitchellh/mapstructure"
+	"github.com/rancher/opni/dagger/config"
+	"github.com/rancher/opni/dagger/helm"
+	"github.com/rancher/opni/dagger/x/cmds"
+	"github.com/spf13/pflag"
+	"golang.org/x/sync/errgroup"
+)
+
+type Builder struct {
+	config.BuilderConfig
+	ctx       context.Context
+	caches    config.Caches
+	cacheMode string
+	client    *dagger.Client
+	sources   *dagger.Directory
+	workdir   string
+}
+
+func main() {
+	if err := run(); err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+}
+
+type runOptions struct {
+	Context context.Context
+	Client  *dagger.Client
+	Args    []string
+}
+
+func run(opts ...runOptions) error {
+	if len(opts) == 0 {
+		opts = append(opts, runOptions{
+			Args: os.Args,
+		})
+	}
+	var debug bool
+	var setup bool
+	var cacheMode string
+	var configs []string
+	var showConfig bool
+	var outputFormat string
+	pf := pflag.NewFlagSet("dagger", pflag.ExitOnError)
+	pf.BoolVar(&debug, "debug", false, "Enable debug logging")
+	pf.StringVar(&cacheMode, "cache-mode", "volumes", "Cache mode (volumes|none)")
+	pf.StringSliceVarP(&configs, "config", "c", nil, "Path to one or more config files")
+	pf.BoolVar(&setup, "setup", false, "Interactive configuration setup")
+	pf.BoolVar(&showConfig, "show-config", false, "Print the final config and exit")
+	pf.StringVarP(&outputFormat, "output-format", "o", "table", "Output format used when --show-config is set (table|json|yaml|toml)")
+	configFlagSet := config.BuildFlagSet(reflect.TypeOf(config.BuilderConfig{}))
```

Comment:
this ends up being very verbose when trying to figure out what dagger run options are available, not sure if there is anything we can do here? Maybe sort these at the end of `--help`

Also the `--help` flag currently errors with 
```
 pflag: help requested
 exit status 2
```
which is a little awkward

Reply:
Added some help text and set the auto generated flags to be hidden, since they are detailed when using --show-config.

Reply:
Fixed :+1: